bentoml/_internal/io_descriptors/base.py

```diff
@@ -83,3 +103,26 @@ async def to_http_response(
         self, obj: IOType, ctx: Context | None = None
     ) -> Response:
         ...
+
+    @overload
+    @abstractmethod
+    async def from_proto(self, request: pb.Part) -> IOType:
+        ...
+
+    @overload
+    @abstractmethod
+    async def from_proto(self, request: pb.Response) -> IOType:
+        ...
+
+    @overload
+    @abstractmethod
+    async def from_proto(self, request: pb.Request) -> IOType:
+        ...
+
+    @abstractmethod
+    async def from_proto(self, request: message.Message) -> IOType:
+        ...
+
+    @abstractmethod
+    async def to_proto(self, obj: IOType) -> MessageMap[str, pb.Part] | message.Message:
```

Comment:
A multipart protobuf message is a MessageMap container.

Reply:
Is that not a subclass of `message.Message`, then?

Reply:
It is not, MessageMap is its one type.

Reply:
I just checked the implementation, it seems like you're just returning a dict? I'm not really sure what's going on.

Reply:
map takes a dictionary as input, however, the return proto for this one is a MesageMap type, which essentially just a dictionary

We can't initialize with MessageMap since it is a internal container and the way they assign value for map are not very nice
```python
map[key].value = "bar"
```

Reply:
That's not a `MessageMap`, then? We should initialize the actual proto object, surely.

Reply:
That is the actual proto object